courses/views.py

```diff
@@ -107,23 +109,37 @@ def edit_module(request, course_id, module_id=None):
         'action_word': 'Add',
         'form': form
     })
+
 def take_quiz(request, quiz_id):
     quiz = Quiz.objects.get(id=quiz_id)
-    questions=quiz.question_set.all()    
+    # Check if Learner already passed the quiz
+    if (QuizResult.objects.filter(learner_id=request.user.id).exists()
+            and QuizResult.objects.filter(quiz_id=quiz.id).exists()):
+        return render(request, 'quiz/take.html', {
+            "passed": True,
+        })
+
+    questions = quiz.question_set.all()
     if request.method == "POST":
-        """
-        To be Implemented
-        form = QuizForm(request.POST, questions)
-        if form.is_valid(): ## Will only ensure the option exists, not correctness.
-            responses = []
+        form = QuizForm(quiz_id, request.POST)
+        if form.is_valid():
```

Comment:
I think we need to add an else branch to this in case the form is not valid https://stackoverflow.com/a/11072097

Reply:
added